### ISSUE TITLE
JavaScript integration from doc does not work

### ISSUE THREAD

**Issue by stephenbe** [ionic-team/stencil#2595]
**I'm submitting a:**

[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**

Nothing happens 
**Expected behavior:**

Loads the test-component

**Steps to reproduce:**

Add this index.html

add this code from the docs : 
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <script type="module">
      import {
        applyPolyfills,
        defineCustomElements,
      } from "https://unpkg.com/test-components/loader";
      applyPolyfills().then(() => {
        defineCustomElements();
      });
    </script>
  </head>
  <body>
    <test-component></test-component>
  </body>
</html>
```
run a simple http-server
I get a 404

**Other information:**

Source :
https://stenciljs.com/docs/javascript

**Comment by adamdbradley** [comment]
This was just an example, and looks like `test-components` actually exists. I guess we could update it to point to ionic instead. Thanks

**Comment by stephenbe** [comment]
Thanks @adamdbradley 

but, I think this issue needs to be opened because this seems to be a broader issue

Here is the code I am trying in a simple index.html with http-server
```
    <script type="module">
      import {
        applyPolyfills,
        defineCustomElements,
      } from "https://cdn.jsdelivr.net/npm/@ionic/core@5.3.1/loader/";
      applyPolyfills().then(() => {
        defineCustomElements();
      });
    </script>
```

This gives me a mime type error :
Failed to load module script: The server responded with a non-JavaScript MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.

If I tried a few files directly inside the loader folder like this
```
    <script type="module">
      import {
        applyPolyfills,
        defineCustomElements,
      } from "https://cdn.jsdelivr.net/npm/@ionic/core@5.3.1/loader/cdn.js";
      applyPolyfills().then(() => {
        defineCustomElements();
      });
    </script>
```


And each file I get this error ( I tried cdn.js, node-main.js, index.cjs.js)
Uncaught SyntaxError: The requested module 'https://cdn.jsdelivr.net/npm/@ionic/core@5.3.1/loader/cdn.js' does not provide an export named 'applyPolyfills'

I also tried this file : index.mjs and I get this error : 

Uncaught TypeError: Failed to resolve module specifier "tslib". Relative references must start with either "/", "./", or "../".

Nothing seems to be working, am I missing something?
